### PR TITLE
media-video/vdr: fix QA issue #695066, remove old ebuilds

### PR DESCRIPTION
**Pull request by madmartin**
Fix misc QA issues:
- USE flag 'permashift_v1' uses reserved underscore character
- UnnecessarySlashStrip in DESTDIR="${D%/}"
- makefile installs pre-compressed manpages

This is Part 1 of a Two-step action.

Step 1 - commit 1: in current media-video/vdr rename useflag to 'permashift'
commit 2: in media-plugin/vdr-permashift rename useflag to 'permashift'
commit 3: media-plugins/vdr-pvr350: remove IUSE yaepg (Closes: https://bugs.gentoo.org/699820 )

wait for stabilize of media-video/vdr on amd64 x86

then I will file another PR - Step 2 - where old ebuilds will be removed:
*    vdr-2.0.6-r1
*    vdr-2.2.0-r3
*    vdr-permashift-1.0.1

This finally closes https://bugs.gentoo.org/695066

### CODE REVIEW

**Comment by madmartin** [comment]
@juippis please review

**Comment by madmartin** [comment]
@mgorny you are the author of bug #695066, so you are invited for review, too

File: media-video/vdr/files/vdr-2.2.0_gentoo.patch [hunk 1]
```diff
@@ -37,31 +36,39 @@ diff -Naur vdr-2.1.7.orig/Makefile vdr-2.1.7/Makefile
...
+@@ -289,9 +287,9 @@
+ install-doc:
+ 	@mkdir -p $(DESTDIR)$(MANDIR)/man1
+ 	@mkdir -p $(DESTDIR)$(MANDIR)/man5
+-	@gzip -c vdr.1 > $(DESTDIR)$(MANDIR)/man1/vdr.1.gz
+-	@gzip -c vdr.5 > $(DESTDIR)$(MANDIR)/man5/vdr.5.gz
+-	@gzip -c svdrpsend.1 > $(DESTDIR)$(MANDIR)/man1/svdrpsend.1.gz
++	@cp vdr.1 $(DESTDIR)$(MANDIR)/man1/
++	@cp vdr.5 $(DESTDIR)$(MANDIR)/man5/
++	@cp svdrpsend.1 $(DESTDIR)$(MANDIR)/man1/
```
Comment (SoapGentoo) [inline media-video/vdr/files/vdr-2.2.0_gentoo.patch]: THANK YOU

File: media-video/vdr/vdr-2.2.0-r4.ebuild [hunk 2]
```diff
@@ -1,12 +1,12 @@
-# Copyright 1999-2019 Gentoo Authors
+# Copyright 2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
 
 inherit eutils flag-o-matic toolchain-funcs
```
Comment (SoapGentoo) [inline media-video/vdr/vdr-2.2.0-r4.ebuild]: do you really still need `eutils` in EAPI 7?

File: media-video/vdr/vdr-2.2.0-r4.ebuild [hunk 3]
```diff
@@ -260,7 +261,7 @@ src_install() {
 	# directory
 	emake install \
 	VIDEODIR="/" \
-	DESTDIR="${D%/}" install || die "emake install failed"
+	DESTDIR="${D}" install || die "emake install failed"
```
Comment (SoapGentoo) [inline media-video/vdr/vdr-2.2.0-r4.ebuild]: `|| die ...` not needed

File: media-video/vdr/files/vdr-2.2.0_gentoo.patch [hunk 4]
```diff
@@ -1,7 +1,6 @@
-diff -Naur vdr-2.1.7.orig/Makefile vdr-2.1.7/Makefile
---- vdr-2.1.7.orig/Makefile	2015-01-18 12:37:45.506034673 +0100
-+++ vdr-2.1.7/Makefile	2015-01-18 12:38:34.086037162 +0100
-@@ -116,7 +116,7 @@
+--- a/Makefile	2019-11-10 12:45:03.370809670 +0100
++++ b/Makefile	2019-11-10 12:48:55.100815904 +0100
```
Comment (SoapGentoo) [inline media-video/vdr/files/vdr-2.2.0_gentoo.patch]: if we're already invalidating the distfiles cache, might as well clean it up properly:
```
--- a/Makefile
+++ b/Makefile
```

File: media-plugins/vdr-pvr350/vdr-pvr350-1.7.5.ebuild [hunk 5]
```diff
@@ -14,27 +14,15 @@ SRC_URI="mirror://vdr-developerorg/${VERSION}/${P}.tgz"
...
 S="${WORKDIR}/${P#vdr-}"
-
-pkg_setup() {
-	vdr-plugin-2_pkg_setup
-
-	if use yaepg; then
-		BUILD_PARAMS="SET_VIDEO_WINDOW=1"
-	fi
-}
+QA_FLAGS_IGNORED="usr/lib/vdr/plugins/libvdr-.* usr/lib64/vdr/plugins/libvdr-.*"
```
Comment (SoapGentoo) [inline media-plugins/vdr-pvr350/vdr-pvr350-1.7.5.ebuild]: put this on two lines for readability
```
QA_FLAGS_IGNORED="
	usr/lib/vdr/plugins/libvdr-.*
	usr/lib64/vdr/plugins/libvdr-.*"

File: media-video/vdr/vdr-2.2.0-r4.ebuild (hunk 2 above)
Comment (madmartin) [reply media-video/vdr/vdr-2.2.0-r4.ebuild]: tested it, indeed not necessary. removed.

File: media-video/vdr/vdr-2.2.0-r4.ebuild (hunk 3 above)
Comment (madmartin) [reply media-video/vdr/vdr-2.2.0-r4.ebuild]: removed

File: media-video/vdr/files/vdr-2.2.0_gentoo.patch (hunk 4 above)
Comment (madmartin) [reply media-video/vdr/files/vdr-2.2.0_gentoo.patch]: lines truncated as requested...

File: media-plugins/vdr-pvr350/vdr-pvr350-1.7.5.ebuild (hunk 5 above)
Comment (madmartin) [reply media-plugins/vdr-pvr350/vdr-pvr350-1.7.5.ebuild]: changed in vdr-permashift and vdr-pvr350

File: media-video/vdr/vdr-2.2.0-r4.ebuild [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+	media-fonts/corefonts
+	bidi? ( dev-libs/fribidi )
+	systemd? ( sys-apps/systemd )"
+
+CONF_DIR=/etc/vdr
+CAP_FILE=${S}/capabilities.sh
+CAPS="# Capabilities of the vdr-executable for use by startscript etc."
+
+pkg_setup() {
+
```
Comment (SoapGentoo) [inline media-video/vdr/vdr-2.2.0-r4.ebuild]: superfluous empty line

File: media-video/vdr/vdr-2.2.0-r4.ebuild [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+	for arg; do
+		CAPS="${CAPS}\n${arg}=1"
+	done
+}
+
+enable_patch() {
+	local arg ARG_UPPER
+	for arg; do
+		ARG_UPPER=$(echo $arg|tr '[:lower:]' '[:upper:]')
+		echo "${ARG_UPPER} = 1" >> Make.config
```
Comment (SoapGentoo) [inline media-video/vdr/vdr-2.2.0-r4.ebuild]: missing `|| die`

File: media-video/vdr/vdr-2.2.0-r4.ebuild [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+
+		# codesnippet to bring the extpng defines into the vdr.pc CXXFLAGS CFLAGS
+		echo "-DUSE_${ARG_UPPER}" >> "${T}"/defines.tmp
+		cat "${T}"/defines.tmp | tr \\\012 ' '  > "${T}"/defines.IUSE
```
Comment (SoapGentoo) [inline media-video/vdr/vdr-2.2.0-r4.ebuild]: both of these commands are missing `|| die`

File: media-video/vdr/vdr-2.2.0-r4.ebuild [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+		export DEFINES_IUSE=$( cat ${T}/defines.IUSE )
+	done
+}
+
+extensions_add_make_conf()
+{
+	# copy all ifdef for extensions-patch to Make.config
+	sed -e '1,/need to touch the following:/d' \
+		-e '/need to touch the following/,/^$/d' \
+		Make.config.template >> Make.config
```
Comment (SoapGentoo) [inline media-video/vdr/vdr-2.2.0-r4.ebuild]: missing `|| die`

File: media-video/vdr/vdr-2.2.0-r4.ebuild [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+}
+
+extensions_all_defines() {
+	# extract all possible settings for extensions-patch
+	sed -e '/^#\?[A-Z].*= 1/!d' -e 's/^#\?//' -e 's/ .*//' \
+		Make.config.template \
+		| sort \
+		| tr '[:upper:]' '[:lower:]'
```
Comment (SoapGentoo) [inline media-video/vdr/vdr-2.2.0-r4.ebuild]: ditto

File: media-video/vdr/vdr-2.2.0-r4.ebuild [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+	api_version=$(awk -F' ' '/define DVB_API_VERSION / {print $3}' "${DVBDIR}"/linux/dvb/version.h)
+	api_version=${api_version}*$(awk -F' ' '/define DVB_API_VERSION_MINOR / {print $3}' "${DVBDIR}"/linux/dvb/version.h)
+
+	if [[ ${api_version:-0} -lt 5*3 ]]; then
+		eerror "DVB header files do not contain s2api support or too old for ${P}"
+		eerror "You cannot compile VDR against old dvb-header"
+		die "DVB headers too old"
+	fi
+
+	cat > Make.config <<-EOT
```
Comment (SoapGentoo) [inline media-video/vdr/vdr-2.2.0-r4.ebuild]: missing `|| die`

File: media-video/vdr/vdr-2.2.0-r4.ebuild (hunk 6 above)
Comment (madmartin) [reply media-video/vdr/vdr-2.2.0-r4.ebuild]: removed

File: media-video/vdr/vdr-2.2.0-r4.ebuild (hunk 7 above)
Comment (madmartin) [reply media-video/vdr/vdr-2.2.0-r4.ebuild]: "die" added

File: media-video/vdr/vdr-2.2.0-r4.ebuild (hunk 8 above)
Comment (madmartin) [reply media-video/vdr/vdr-2.2.0-r4.ebuild]: "die" added

File: media-video/vdr/vdr-2.2.0-r4.ebuild (hunk 10 above)
Comment (madmartin) [reply media-video/vdr/vdr-2.2.0-r4.ebuild]: "die" added

File: media-video/vdr/vdr-2.2.0-r4.ebuild (hunk 9 above)
Comment (madmartin) [reply media-video/vdr/vdr-2.2.0-r4.ebuild]: "die" added

File: media-video/vdr/vdr-2.2.0-r4.ebuild (hunk 11 above)
Comment (madmartin) [reply media-video/vdr/vdr-2.2.0-r4.ebuild]: "die" added

File: media-video/vdr/vdr-2.2.0-r4.ebuild [hunk 12]
```diff
@@ -0,0 +1,302 @@
...
+IUSE="bidi debug +kbd html systemd vanilla ${EXT_PATCH_FLAGS} ${EXT_PATCH_FLAGS_RENAMED}"
+
+COMMON_DEPEND="virtual/jpeg:*
+	sys-libs/libcap
+	>=media-libs/fontconfig-2.4.2
+	>=media-libs/freetype-2"
+
+DEPEND="${COMMON_DEPEND}
+	>=virtual/linuxtv-dvb-headers-5.3
+	sys-devel/gettext"
```
Comment (SoapGentoo) [inline media-video/vdr/vdr-2.2.0-r4.ebuild]: `gettext` likely only belongs in `BDEPEND`

Comment (madmartin) [reply media-video/vdr/vdr-2.2.0-r4.ebuild]: added `BDEPEND="sys-devel/gettext" ` as requested

**Pull request closed without merging**